tools/base/envoy_python.bzl

```diff
@@ -126,3 +127,115 @@ def envoy_py_script(
         envoy_prefix = envoy_prefix,
         test = False,
     )
+
+def _py_data(
+        name,
+        source,
+        filters = [],
+        visibility = ["//visibility:public"],
+        loader = "json.loads",
+        imports = ["json"],
+        deps = [],
+        envoy_prefix = "@envoy"):
+    py_file = "%s.py" % name.split(".")[-1]
+
+    # Add python imports for filters, transforms python deps labels for workspace
+    # rules into python importable dotted notation.
+    filter_imports = []
+    for filter in filters:
+        if filter.startswith("@envoy"):
+            filter = filter[6:]
+        elif filter.startswith("@"):
+            filter = filter[1:].replace("//", ".")
+        filter_imports.append(filter.strip("//").replace("/", ".").replace(":", "."))
+    filter_imports = "\\n".join([
+        "from %s import main as _filter%s" %
+        (filter, i)
+        for i, filter in enumerate(filter_imports)
+    ])
+
+    # Add filter aliases that can be run
+    filter_tuple = ", ".join(["_filter%s" % i for i in range(0, len(filters))])
+    filter_tuple = filter_tuple or "()"
+    imports = "\\n".join([("import %s" % imp) for imp in imports] + [""]) + filter_imports
+    template_rule = "%s//tools/base:base_load_data.py" % envoy_prefix
+    template = "$(location %s)" % template_rule
+    native.genrule(
+        name = "py_script_%s" % py_file,
+        cmd = (
+            "sed 's#__DATA_FILE__#%s#; s#__IMPORTS__#%s#; s#__LOADER__#%s#; s#__FILTERS__#%s#' %s > $@" %
+            (source.split(":")[1], imports, loader, filter_tuple, template)
+        ),
+        tools = [template_rule, source],
+        outs = [py_file],
+    )
```

Comment:
This seems really complex. I don't get why `py_data` isn't a simple templated import of a Bazel dictionary?

Reply:
its not necessarily a dictionary and it needs to load both yaml and json, so it has a loader in python

Reply:
(potentially this could also load/mangle protobuf im thinking)